.github/workflows/check-modified-assets.yml

```diff
@@ -30,15 +43,29 @@ jobs:
         runs-on: ubuntu-latest
         steps:
             - uses: actions/checkout@v3
+
+            - name: Cache node_modules
+              id: cache-node-modules
+              uses: actions/cache@v3
+              env:
+                  cache-name: cache-node-modules
+              with:
+                  path: node_modules
+                  key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+
             - name: Setup node version and npm cache
               uses: actions/setup-node@v3
               with:
                   node-version-file: '.nvmrc'
                   cache: 'npm'
-            - name: npm install
-              run: |
-                  npm ci --no-optional
-                  npm run build:check-assets
+
+            - name: Install Node Dependencies
+              if: steps.cache-node-modules.outputs.cache-hit != 'true'
+              run: npm ci --no-optional
```

Comment:
>@alexflorisca, is there a value in keeping the[ caching and restoring dependencies functionality ](https://github.com/actions/setup-node#caching-global-packages-data)from the setup-node GitHub action (e.g. cache: 'npm') after using the actions/cache to cache the entire node_modules folder?

Yes, they do slightly different things. Cachine node_modules is useful for not having to do `npm ci` at all. But when we change a package version, we will need to `npm ci`. This is where the npm cache is useful, as it will grab whatever packages haven't changed from a local cache, and it will only download the ones that have changed, saving bandwidth. So they work together well

Reply:
Yep, you're right, I've removed them here, thanks!

Reply:
We need these here as we are running npm install and build on two different branches. The first job runs on trunk, and the second one runs on the PR branch in order to compare assets.

Reply:
@alexflorisca, wouldn't the two branches share the same `node_modules` after running the first `npm ci` since `node_modules` is ignored by git?